mesonbuild/msetup.py

```diff
@@ -149,13 +149,6 @@ def validate_dirs(self, dir1, dir2, reconfigure, wipe):
                 sys.exit(1)
         return src_dir, build_dir
 
-    def check_pkgconfig_envvar(self, env):
-        curvar = os.environ.get('PKG_CONFIG_PATH', '')
-        if curvar != env.coredata.pkgconf_envvar:
-            mlog.warning('PKG_CONFIG_PATH has changed between invocations from "%s" to "%s".' %
-                         (env.coredata.pkgconf_envvar, curvar))
-            env.coredata.pkgconf_envvar = curvar
-
```

Comment:
I don't see a replacement for this, is it intentional to stop warning about `PKG_CONFIG_PATH` changing between invocations when it's not specified explicitly on the command-line?

Reply:
Yes, this is intentional. We don't take it into account after the initial configuration anymore, just like CFLAGS. Is it worth keeping the warning?

Reply:
We don't take it into account, but we should definitely warn about it in case the user tries to change the env var and gets surprised. We should change the warning to say that the user should use the command-line option instead.

Reply:
To clarify: we should say that if the user wants to change the value, they should use the command-line option.

Reply:
that's fair, I'll make that change.

Reply:
Although, looking at the code, that would really belong in mconf.py, and we should probably do that for all environment varialbes (CFLAGS, LDFLAGS, CC, etc), and that feels like a separate change.

Reply:
As long as you promise to add that too, I don't mind ;)

Reply:
I've created #5223 and assigned it to myself.